.github/workflows/generate-crd.yml

```diff
@@ -11,40 +14,103 @@ jobs:
+      - name: Determine docs branch
+        id: set_docs_branch
+        shell: bash
+        run: |
+          TAG="${OPERATOR_TAG}"
+          echo "Operator tag: $TAG"
+
+          if [[ "$TAG" == *"v2.4"* ]]; then
+            echo "Operator tag contains v2.4 - using docs branch v/24.3."
+            if git ls-remote --exit-code --heads https://github.com/redpanda-data/docs.git v/24.3; then
+              DOCS_BRANCH="v/24.3"
+            else
+              echo "Error: Docs branch v/24.3 not found."
+              exit 1
+            fi
+          else
+            # Fetch the latest docs version from the antora.yml file on the main docs branch.
+            ANTORA_VERSION=$(curl -s https://raw.githubusercontent.com/redpanda-data/docs/main/antora.yml | grep '^version:' | sed -E 's/version: *//')
+            echo "Version in docs main branch: $ANTORA_VERSION"
+            # If the operator tag starts with that version, then commit to main.
+            if [[ "$TAG" == "v${ANTORA_VERSION}"* ]]; then
+              DOCS_BRANCH="main"
+            else
+              # Derive the docs branch name from the operator tag.
+              BASE_VERSION=$(echo "$TAG" | sed -E 's/^v([0-9]+\.[0-9]+).*/\1/')
+              DOCS_BRANCH="v/${BASE_VERSION}"
+              if ! git ls-remote --exit-code --heads https://github.com/redpanda-data/docs.git "$DOCS_BRANCH"; then
+                echo "Error: Docs branch $DOCS_BRANCH not found."
+                exit 1
+              fi
+            fi
+          fi
+
+          echo "DOCS_BRANCH=$DOCS_BRANCH" >> $GITHUB_ENV
+          echo "Using docs branch: $DOCS_BRANCH"
+
+      # Checkout the operator repository at the specified tag.
+      - name: Checkout redpanda operator repository
         uses: actions/checkout@v4
         with:
           repository: redpanda-data/redpanda-operator
-          ref: main
+          ref: ${{ env.OPERATOR_TAG }}
           path: redpanda
           token: ${{ env.ACTIONS_BOT_TOKEN }}
-      # Checkout the redpanda-docs repository based on the FORMATTED_BRANCH.
-      - name: Checkout this repository
+
+      # Checkout the docs repository using the determined docs branch.
+      - name: Checkout redpanda-docs repository
         uses: actions/checkout@v4
         with:
-          ref: main
           repository: redpanda-data/docs
+          ref: ${{ env.DOCS_BRANCH }}
           path: redpanda-docs
           token: ${{ env.ACTIONS_BOT_TOKEN }}
-      # Download crd-ref-docs utility for doc generation.
+
       - name: Download crd-ref-docs
         run: |
           curl -fLO https://github.com/elastic/crd-ref-docs/releases/download/v0.1.0/crd-ref-docs_0.1.0_Linux_x86_64.tar.gz
           tar -xzf crd-ref-docs_0.1.0_Linux_x86_64.tar.gz
           chmod +x crd-ref-docs
           sudo mv crd-ref-docs /usr/local/bin/
-      # Generate Operator CRD documentation.
-      - name: Generate Operator CRD Docs
+
+      - name: Download helm-docs
```

Comment:
The README.md in the redpanda chart should always be up to date. It doesn't look like there's a need for docs to run this?